app/models/policy_host.rb

```diff
@@ -8,10 +8,25 @@ class PolicyHost < ApplicationRecord
   validates :policy, presence: true
   validates :host, presence: true, on: :create
   validates :host_id, presence: true, uniqueness: { scope: :policy }
+  validate :host_supported?, on: :create
 
   def self.import_from_policy(policy_id, host_ids)
     import(host_ids.map do |host_id|
       { host_id: host_id, policy_id: policy_id }
     end)
   end
+
+  def host_supported?
+    profile = Profile.find_by(policy: policy)
+    errors.add(:host, 'Unsupported OS major version') unless host&.os_major_version.to_s ==
+                                                             profile&.os_major_version.to_s
+
+    errors.add(:host, 'Unsupported OS minor version') unless supported_os_minor_version(profile)
+  end
+
+  def supported_os_minor_version(profile)
+    ssg_supported_versions = SupportedSsg.ssg_versions_for_os(host&.os_major_version, host&.os_minor_version).uniq.sort
+    supported_versions_intersect = ((profile&.supported_benchmark_versions || []) & ssg_supported_versions)&.sort
+    supported_versions_intersect == ssg_supported_versions
```

Comment:
This seems to me a bit suspicious, first of all you have a lot of `uniq` ans `sort` calls everywhere and the `|| []` should not be necessary. Maybe you can use the `ssg_supported_versions` when constructing the list of `supported_benchmarks` and just check if the result is empty. I need to look into this more, but my guess is that the complexity can be lowered here significantly.

Reply:
Yeah, this was a little strange because sometimes `profile` is nil so `profile&.supported_benchmark_versions` would return nil and then this would mess up the intersection of `profile&.supported_benchmark_versions & ssg_supported_versions` because instersecting `nil & [..]` returns `false`
But I see what you mean.. I think I can make this less complex by doing something like what you suggested.